be/src/block_cache/starcache/aux_funcs.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "block_cache/starcache/aux_funcs.h"
+
+#include <butil/crc32c.h>
+
+namespace starrocks::starcache {
+
+uint32_t crc32(const butil::IOBuf& buf) {
+    uint32_t hash = 0;
+    const size_t block_num = buf.backing_block_num();
+    for (size_t i = 0; i < block_num; ++i) {
+        auto sp = buf.backing_block(i);
+        if (!sp.empty()) {
+            hash = butil::crc32c::Extend(hash, sp.data(), sp.size());
+        }
+    }
+    return hash;
+}
+
+uint64_t cachekey2id(const std::string& key) {
+    return hash64(key.data(), key.size(), 0);
+}
+
+size_t align_buf(const butil::IOBuf& buf, void** aligned_data) {
```

Comment:
what's purpose of this function?

Reply:
> what's purpose of this function?

When using direct io, the buffer address and size to write or read need to be aligned by the disk parameter. This function is used to finish it.